src/ru/desu/src/eu/kanade/tachiyomi/extension/ru/desu/Desu.kt

```diff
@@ -206,14 +205,16 @@ class Desu : HttpSource() {
         val objChapter = obj["chapters"]!!
         return objChapter.jsonObject["list"]!!.jsonArray.map {
             val chapterObj = it.jsonObject
-            val ch = chapterObj["ch"]!!.jsonPrimitive.float
-            val fullNumStr = "${chapterObj["vol"]!!.jsonPrimitive.int}. Глава " + DecimalFormat("#,###.##").format(ch).replace(",", ".")
+            val ch = chapterObj["ch"]!!.jsonPrimitive.content
+            val vol = chapterObj["vol"]!!.jsonPrimitive.content
+            val fullNumStr = "$vol. Глава $ch"
             val title = chapterObj["title"]!!.jsonPrimitive.contentOrNull ?: ""
 
             SChapter.create().apply {
                 name = "$fullNumStr $title"
-                url = "/$cid/chapter/${chapterObj["id"]!!.jsonPrimitive.int}"
-                chapter_number = ch
+                // #apiChapter - JSON API url to automatically delete when chapter is opened in browser
+                url = "/manga/$cid/vol$vol/ch$ch/rus" + "#apiChapter/$cid/chapter/${chapterObj["id"]!!.jsonPrimitive.int}"
+                chapter_number = ch.toFloat()
```

Comment:
This will fail if for some reason the string doesn't have a valid float number.

```suggestion
                chapter_number = ch.toFloatOrNull() ?: -1f
```